homeTask2/src/components/UserDetails.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { UserItem } from './types'
+import UserForm from './UserForm'
+
+type Props = {
+    userHistoryList: UserItem[]
+    editUser: (id: number, userInfo: UserItem) => void
+}
+const UserDetails = ({userHistoryList, editUser}: Props) => {
+    const {userId} = useParams<{userId: string}>();
+    const [isEditing, setEditing] = useState(false)
+    const user = userHistoryList.find(item => item.id.toString() === userId)
+    if(user === undefined){
+        return <p>User not found</p>
+    }
+    const { id, firstName } = user
+    const onEdit = (id: number, userInfo: UserItem) => {
+        editUser(userId, userInfo)
```

Comment:
use `id` value from `user` here, to pass a number instead of string from query params